Project in npm registry

Related to this issue [https://github.com/playcanvas/playcanvas-sync/issues/30](https://github.com/playcanvas/playcanvas-sync/issues/30)

We currently use this project as an npm module installed via the direct GitHub link. This causes deployment issues with CI/CD pipelines on npm v7 and later, where the deployment server requires an SSH key for public repositories. This is a known issue tracked [here](https://github.com/npm/cli/issues/2610).

I published this `playcanvas-sync` repository to the npm registry without realizing it wouldn't associate the project with the Github owners. In hindsight that does make sense. I've invited the 'playcanvas' account to become a maintainer and eventual owner of the [npm package](https://www.npmjs.com/package/playcanvas-sync), but wanted to open this line of dialogue here in case the maintainers would prefer alternative actions such as renaming the package or removing it altogether.

@yaustar Yes. I think having official NPM module would be nice.

playanvas-sync is defined as devDependency in package.json:

```
{
  . . .
  "devDependencies": {
    . . .
    "playcanvas-sync": "https://github.com/playcanvas/playcanvas-sync.git#1f3c179fb96a9efe4ba8e0f78638193db5d17949",
    . . .
  },
  . . .
}

```

And CI/CD pipelines return next:

```
[4] npm ERR! Error while executing:
[5] npm ERR! /usr/bin/git ls-remote -h -t ssh://[USER]@github.com/playcanvas/playcanvas-sync.git
[6] npm ERR! 
[7] npm ERR! Warning: Permanently added the ECDSA host key for IP address [IP] to the list of known hosts.
[8] npm ERR! Permission denied (publickey).
[9] npm ERR! fatal: Could not read from remote repository.
[10] npm ERR! 
[11] npm ERR! Please make sure you have the correct access rights
[12] npm ERR! and the repository exists.
[13] npm ERR! 
[14] npm ERR! exited with error code: 128
```

--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -1,0 +1,20 @@
+name: Publish Package to npmjs
+on:
+  release:
+    types: [published]
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v3
+        with:
+          node-version: '18.x'
+          registry-url: 'https://registry.npmjs.org'
+      - run: npm ci
+      # TODO: Add back in after linting issues have been resolved.
+      # - run: npm lint
+      - run: npm version ${{ github.event.release.tag_name }} --no-git-tag-version
+      - run: npm publish
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}

--- a/bin/pcsync.js
+++ b/bin/pcsync.js
@@ -1,10 +1,10 @@
 #!/usr/bin/env node
 
-const OverwriteAllLocalWithRemote = require('./src/sync-commands/overwrite-all-local-with-remote');
-const OverwriteAllRemoteWithLocal = require('./src/sync-commands/overwrite-all-remote-with-local');
-const CUtils = require('./src/utils/common-utils');
-const SyncUtils = require('./src/sync-commands/sync-utils');
-const SCUtils = require('./src/sync-commands/sync-command-utils');
+const OverwriteAllLocalWithRemote = require('../src/sync-commands/overwrite-all-local-with-remote');
+const OverwriteAllRemoteWithLocal = require('../src/sync-commands/overwrite-all-remote-with-local');
+const CUtils = require('../src/utils/common-utils');
+const SyncUtils = require('../src/sync-commands/sync-utils');
+const SCUtils = require('../src/sync-commands/sync-command-utils');
 
 const program = require('commander');
 

--- a/bin/pcwatch.js
+++ b/bin/pcwatch.js
@@ -1,15 +1,15 @@
 #!/usr/bin/env node
 
-const GetConfig = require('./src/utils/get-config');
+const GetConfig = require('../src/utils/get-config');
 const program = require('commander');
 
-const CUtils = require('./src/utils/common-utils');
-const WatchUtils = require('./src/watch-actions/watch-utils');
-const ActionCreated = require('./src/watch-actions/action-created');
-const SyncUtils = require('./src/sync-commands/sync-utils');
-const LocalWatcher = require('./src/utils/local-watcher');
-const CacheUtils = require('./src/utils/cache-utils');
-const LocalTraversal = require('./src/utils/local-traversal');
+const CUtils = require('../src/utils/common-utils');
+const WatchUtils = require('../src/watch-actions/watch-utils');
+const ActionCreated = require('../src/watch-actions/action-created');
+const SyncUtils = require('../src/sync-commands/sync-utils');
+const LocalWatcher = require('../src/utils/local-watcher');
+const CacheUtils = require('../src/utils/cache-utils');
+const LocalTraversal = require('../src/utils/local-traversal');
 
 program.option('-f, --force', 'skip local/remote equality check');
 

--- a/package.json
+++ b/package.json
@@ -41,7 +41,7 @@
     "eslint": "^8.24.0"
   },
   "scripts": {
-    "lint": "./node_modules/.bin/eslint --ext .js,.mjs src bin"
+    "lint": "eslint --ext .js,.mjs src bin"
   },
   "bin": {
     "pcsync": "./bin/pcsync.js",

--- a/package.json
+++ b/package.json
@@ -9,6 +9,11 @@
     "sync",
     "local"
   ],
+  "files": [
+    "bin",
+    "docs",
+    "src"
+  ],
   "license": "MIT",
   "main": "index.js",
   "bugs": {
@@ -36,10 +41,10 @@
     "eslint": "^8.24.0"
   },
   "scripts": {
-    "lint": "eslint --ext .js,.mjs src pcsync.js pcwatch.js"
+    "lint": "./node_modules/bin/eslint --ext .js,.mjs src bin"
   },
   "bin": {
-    "pcsync": "./pcsync.js",
-    "pcwatch": "./pcwatch.js"
+    "pcsync": "./bin/pcsync.js",
+    "pcwatch": "./bin/pcwatch.js"
   }
 }

--- a/package.json
+++ b/package.json
@@ -41,7 +41,7 @@
     "eslint": "^8.24.0"
   },
   "scripts": {
-    "lint": "./node_modules/bin/eslint --ext .js,.mjs src bin"
+    "lint": "./node_modules/.bin/eslint --ext .js,.mjs src bin"
   },
   "bin": {
     "pcsync": "./bin/pcsync.js",

